qiskit/providers/aer/jobs/aerjobset.py

```diff
@@ -218,6 +232,63 @@ def _get_worker_result(self, worker: int, timeout: Optional[float] = None):
                     "Timeout while waiting for JobSet results")
         return result
 
+    def _merge_results(self, result_list: List[Result]):
+        if len(result_list) == 1:
+            return result_list[0]
+
+        master_result = result_list[0].to_dict()
+        sub_result = result_list[1].to_dict()
+        result_list = []
+
+        for (_master_result, _sub_result) in zip(master_result["results"], sub_result["results"]):
+            result_list.append(self._merge_exp(_master_result, _sub_result))
+        master_result["results"] = result_list
+        return Result.from_dict(master_result)
+
+    def _merge_experiments_result(self, results: List[Result]):
+        """Merge all experiments into a single in a`Result`
+
+        this function merges the counts and the number of shots
+        from each experiment in a `Result` for a noise simulation
+        if `id` in metadata field is the same.
+
+        Args:
+            results: Result list whose experiments will be combined.
+
+        Returns:
+            list: Result list
+
+        """
+        results_list = []
+        for each_result in results:
+            _merge_results = []
+            master_id = None
+            master_result = None
+
+            _dict_result = each_result.to_dict()
```

Comment:
Again, don't convert to dicts. Either apply this to the dicts before they are converted to results, or use the Result attrbiutes.

Reply:
Remove dict function.